### PR TITLE
docs(*): document the breaking change introduced in e1da4be

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs update.

**What is the current behavior? (You can also link to an open issue here)**
Although e1da4be introduced a breaking change, it wasn't documented as such.

**What is the new behavior (if this is a feature change)?**
The breaking change is documented.

**Does this PR introduce a breaking change?**
No.
<sub>(In a way. "Let me introduce you this breaking change. It has been around for 2 releases, but I don't think you've met.)"</sub>

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -270,12 +270,14 @@ Please read the [Sandbox Removal Blog Post](http://angularjs.blogspot.com/2016/0
...
 - **ngModelOptions:** allow options to be inherited from ancestor `ngModelOptions` ([87a2ff](https://github.com/angular/angular.js/commit/87a2ff76af5d0a9268d8eb84db5755077d27c84c) [#10922](https://github.com/angular/angular.js/issues/10922))
-- **input:** add support for binding to `input[type=range]` ([913016](https://github.com/angular/angular.js/commit/9130166767c4792c5d32d08a918fc7becf32c9a6) [#5892](https://github.com/angular/angular.js/issues/5892) [#14870](https://github.com/angular/angular.js/issues/14870))
+- **input:**
+  - add support for binding to `input[type=range]` ([913016](https://github.com/angular/angular.js/commit/9130166767c4792c5d32d08a918fc7becf32c9a6) [#5892](https://github.com/angular/angular.js/issues/5892) [#14870](https://github.com/angular/angular.js/issues/14870))
+  - add support for `step` to `input[type=number]` ([e1da4be](https://github.com/angular/angular.js/commit/e1da4bed8e291003d485a8ad346ab80bed8ae2e3) [#10597](https://github.com/angular/angular.js/issues/10597))
```
Comment (petebacondarwin) [inline CHANGELOG.md]: I think that these two 913016 and e1da4be should be features.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1286,6 +1288,38 @@ to a valid value, the input will in turn set the model to this new valid value v
...
+
+Number inputs that use `ngModel` and specify a `step` constraint (via `step`/`ngStep` attributes)
+will now have a new validator (`step`), which will verify that the current value is valid under the
+`step` constraint (according to the [spec](https://www.w3.org/TR/html5/forms.html#the-step-attribute)).
+Previously, the `step` constraint was ignored by `ngModel`, treating values as valid even when there
+was a step-mismatch.
+
+If you want to restore the previous behavior (disabling `step` validation), but still want to be
+able to use the `step` attribute, you can overwrite the built-in `step` validator by using a custom
+directive. For example:
```
Comment (petebacondarwin) [inline CHANGELOG.md]: How about?

> If you want to restore the previous behavior (use the step attribute while disabling step validation), you can overwrite ...

**Comment by petebacondarwin** [comment]
Good work @gkalpak as usual you put me to shame by being much more fastidious than me :-)

File: CHANGELOG.md (hunk 1 above)
Comment (gkalpak) [inline CHANGELOG.md]: They are :smiley: Tricked by the shortened GitHub diff?

**Comment by gkalpak** [comment]
Nah! We're rather taking turns at that :smiley: 
(Although you do use fancier words 😛)

**Pull request merged**